### PR TITLE
Changes to race ethnicity disaggregation on data page

### LINKED ISSUE #701: Changes to Race/Ethnicity disaggregation on Data page
## Problem

Currently, the Data Viz page doesn't handle the "Race/Ethnicity" disaggregation very well. There are so many categories that we run out of colors for the legend!

And the tooltip overflows:

## Solution

The solution we've decided on is to reduce the number of categories to just these:

- American Indian or Alaska Native (non-Hispanic or Unknown ethnicity) 
- Asian (non-Hispanic or Unknown ethnicity) 
- Black (non-Hispanic or Unknown ethnicity) 
- Hispanic or Latino (any race) 
- Native Hawaiian or Pacific Islander (non-Hispanic or Unknown ethnicity) 
- White (non-Hispanic or Unknown ethnicity) 
- More than one race (non-Hispanic or Unknown ethnicity) 
- Other (non-Hispanic or Unknown ethnicity) 
- Unknown (non-Hispanic or Unknown ethnicity) 

We'll have to group the existing datapoints into one of these categories, and then sum up the value of all datapoints in that category. (The example below will make this clearer.)

The grouping logic should be as follows:

- American Indian or Alaska Native (non-Hispanic or Unknown ethnicity) should be the sum of:
    - American Indian or Alaska Native / Unknown Ethnicity
    - American Indian or Alaska Native / Not Hispanic or Latino
- Asian (non-Hispanic or Unknown ethnicity) should be the sum of:
    - Asian / Unknown Ethnicity
    - Asian / Not Hispanic or Latino
- Black (non-Hispanic or Unknown ethnicity) should be the sum of:
    - Black / Unknown Ethnicity
    - Black / Not Hispanic or Latino
- Hispanic or Latino (any race) should be the sum of:
     - American Indian or Alaska Native / Hispanic or Latino
     - Asian / Hispanic or Latino
     - Black / Hispanic or Latino
     - Native Hawaiian or Pacific Islander / Hispanic or Latino
     - White / Hispanic or Latino
     - More than one race / Hispanic or Latino
     - Other / Hispanic or Latino
     - Unknown / Hispanic or Latino
- Native Hawaiian or Pacific Islander (non-Hispanic or Unknown ethnicity) should be the sum of:
    - Native Hawaiian or Pacific Islander / Unknown Ethnicity
    - Native Hawaiian or Pacific Islander / Not Hispanic or Latino
- White (non-Hispanic or Unknown ethnicity) should be the sum of:
    - White / Unknown Ethnicity
    - White / Not Hispanic or Latino
- More than one race (non-Hispanic or Unknown ethnicity) should be the sum of:
    - More than one race / Unknown Ethnicity
    - More than one race / Not Hispanic or Latino
- Other (non-Hispanic or Unknown ethnicity) should be the sum of:
   - Other / Unknown Ethnicity
   - Other / Not Hispanic or Latino
- Unknown (non-Hispanic or Unknown ethnicity) should be the sum of:
   - Unknown / Unknown Ethnicity
   - Unknown / Not Hispanic or Latino
  
## Example

For instance, suppose the backend sent the following datapoints:

```
    {
      "metric_definition_key": "JAILS_PRE_ADJUDICATION_POPULATION",
      "disaggregation_display_name": "Race / Ethnicity",
      "dimension_display_name": "White / Unknown Ethnicity",
      "value": 5
    },
    {
      "metric_definition_key": "JAILS_PRE_ADJUDICATION_POPULATION",
      "disaggregation_display_name": "Race / Ethnicity",
      "dimension_display_name": "White / Not Hispanic or Latino",
      "value": 2
    },
    {
      "metric_definition_key": "JAILS_PRE_ADJUDICATION_POPULATION",
      "disaggregation_display_name": "Race / Ethnicity",
      "dimension_display_name": "White / Hispanic or Latino",
      "value": 9
    },
    {
      "metric_definition_key": "JAILS_PRE_ADJUDICATION_POPULATION",
      "disaggregation_display_name": "Race / Ethnicity",
      "dimension_display_name": "Asian / Hispanic or Latino",
      "value": 1
    },
```

We would combine the first two datapoints into the new "White (non-Hispanic or Unknown ethnicity)" category and give it a value of 5 + 2 = 7. We would group the second two datapoints into the new "Hispanic or Latino (any race)" category and give it a value of 9+1 = 10.

Basically, we should act as if the backend sent us this response instead:

```
    {
      "metric_definition_key": "JAILS_PRE_ADJUDICATION_POPULATION",
      "disaggregation_display_name": "Race / Ethnicity",
      "dimension_display_name": "White (non-Hispanic or Unknown ethnicity)",
      "value": 7
    },
    {
      "metric_definition_key": "JAILS_PRE_ADJUDICATION_POPULATION",
      "disaggregation_display_name": "Race / Ethnicity",
      "dimension_display_name": "Hispanic or Latino (any race)",
      "value": 10
    },
```

## How?

It's totally up to you, but I would suggest putting this logic in the `datapointsByMetric` method in BaseDatapointStore.ts. We can probably combine it with the existing `reduce` statement if we make a few changes. As we loop through datapoints, if we come across a datapoint with `disaggregation_display_name` = "Race / Ethnicity", we should look at its `dimension_display_name` (e.g. Asian / Hispanic or Latino) and figure out which of the new dimensions it belongs to (e.g. Hispanic or Latino (any race)). If we've already seen any datapoints for that new dimension for this time period, (e.g. we already say White / Hispanic or Latino) we should add the values together. Otherwise, we just start with this datapoints value.

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Union of race/ethnicity's dimension names

## Type of change
Type: Non-breaking refactor

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #701 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/DataViz/ConnectedDatapointsView.tsx [hunk 1]
```diff
@@ -61,14 +64,29 @@ const ConnectedDatapointsView: React.FC<{
...
 
+  let dimensionNamesByDisaggregation =
+    datapointsStore.dimensionNamesByMetricAndDisaggregation[metric];
+
+  if (
+    Object.keys(
+      datapointsStore.dimensionNamesByMetricAndDisaggregation[metric]
+    ).includes("Race / Ethnicity")
+  ) {
+    const raceEthnicityDimensionNames = Array.from(
```
Comment (lilidworkin) [inline publisher/src/components/DataViz/ConnectedDatapointsView.tsx:75]: Is it possible to sort this array so it's in this order:

Hispanic or Latino (any race)
American Indian ...
Asian ...
Black ...
More than one ...
Native Hawaiian ...
White ...
Other ...
Unknown ...

Thank you!

**Review by lilidworkin — APPROVED** [review]
This looks fantastic @ivan-kishko ! You came up with such a simple and elegant solution! I tested it out and it looks like it's working great. I left one request about sorting the strings, and then it's good to merge!

File: common/stores/BaseDatapointsStore.ts [hunk 2]
```diff
@@ -96,6 +97,36 @@ abstract class DatapointsStore {
...
           ] = {};
         }
+
+        const dimensionName = Object.keys(UnitedRaceEthnicityKeys).includes(
+          dp.dimension_display_name
+        )
+          ? UnitedRaceEthnicityKeys[dp.dimension_display_name]
+          : dp.dimension_display_name;
+
+        const isDimensionNameExists =
```
Comment (mxosman) [inline common/stores/BaseDatapointsStore.ts:107]: tiny nit: maybe `hasDimensionName`?

**Review by mxosman — APPROVED** [review]
Nice solution, Ivan! LGTM as well!!

File: publisher/src/components/DataViz/ConnectedDatapointsView.tsx (hunk 1 above)
Comment (ivan-kishko) [reply publisher/src/components/DataViz/ConnectedDatapointsView.tsx:75]: [45d82ce](https://github.com/Recidiviz/justice-counts/pull/711/commits/45d82ce3f426716ec1745c542b76ebd7ce5f535a)

**Review by lilidworkin — APPROVED** [review]
Great, thanks!

**Pull request merged**